File: src/main/java/studio/karllang/karl/errors/RuntimeError/RuntimeError.java
package studio.karllang.karl.errors.RuntimeError;

import studio.karllang.karl.errors.Error;

/** Represents a runtime error in Karl source code. */
public class RuntimeError extends Error {
  public RuntimeError(String message, String filePath, int line, int position) {
    super("Runtime Error", message, filePath, line, position);
  }
}

File: src/main/kotlin/studio/karllang/karl/errors/runtime/RuntimeError.kt
package studio.karllang.karl.errors.runtime

open class RuntimeError(
    private val msg: String,
    private val pos: Int,
    private val line: Int,
    private val lineString: String
) : Throwable() {
    fun getMsg(): String {
        return msg
    }

    fun getPosition(): Int {
        return pos
    }

    fun getLine(): Int {
        return line
    }

    fun getLineString(): String {
        return lineString
    }
}